src/airflow/Dockerfile

```
@@ -1,11 +1,10 @@
-FROM apache/airflow:2.7.2-python3.10
+FROM apache/airflow:slim-2.7.2-python3.9
 
 # Install additional Python requirements.
 # --no-cache-dir is a good practice when installing packages using pip, because it helps to keep the image lightweight.
 COPY requirements.txt /requirements.txt
-RUN pip install --user --upgrade pip
-RUN pip install --no-cache-dir --user -r /requirements.txt # --no-cache-dir good practise when installing packages using pip. It helps to keep the image lightweight
-
+RUN pip install --user --upgrade pip && \
+    pip install --no-cache-dir --user -r /requirements.txt
```

```suggestion
RUN pip install --quiet --user --no-cache-dir --upgrade pip setuptools && \
    pip install --quiet --user --no-cache-dir -r /requirements.txt
```

* Using `--quiet --user --no-cache-dir` for both commands
* When upgrading `pip`, it's usually a good idea to upgrade `setuptools` as well. Together, their update usually solves all installation problems